Source/AaxDecrypter/NetworkFileStream.cs

```diff
@@ -151,39 +159,82 @@ public async Task BeginDownloadingAsync()
 			if (ContentLength != 0 && WritePosition > ContentLength)
 				throw new WebException($"Specified write position (0x{WritePosition:X10}) is larger than  {nameof(ContentLength)} (0x{ContentLength:X10}).");
 
+			//Initiate connection with the first request block and
+			//get the total content length before returning.
+			using var client = new HttpClient();
+			var response = await RequestNextByteRangeAsync(client);
+
+			if (ContentLength != 0 && ContentLength != response.FileSize)
+				throw new WebException($"Content length of 0x{response.FileSize:X10} differs from partially downloaded content length of 0x{ContentLength:X10}");
+			
+			ContentLength = response.FileSize;
+
+			_downloadedPiece = new EventWaitHandle(false, EventResetMode.AutoReset);
+			//Hand off the open request to the downloader to download and write data to file.
+			DownloadTask = Task.Run(() => DownloadLoopInternal(response), _cancellationSource.Token);
+		}
+
+		private async Task DownloadLoopInternal(BlockResponse initialResponse)
+		{
+			await DownloadToFile(initialResponse);
+			initialResponse.Dispose();
+
+			try
+			{
+				using var client = new HttpClient();
+				while (WritePosition < ContentLength && !IsCancelled)
+				{
+					using var response = await RequestNextByteRangeAsync(client);
+					await DownloadToFile(response);
+				}
+			}
```

Comment:
Unlike the ADRM download links which can be reliably downloaded in their entirety in a single Http connection, the widevine links tend to close the connection prematurely. Upon inspecting the Audible app's network traffic, I discovered that the App will only download 64MB chunks at a time. I changed the network stream to do the same.